client/lib/titan/new-mailbox.js

```diff
@@ -226,7 +259,46 @@ const transformMailboxForCart = ( {
 	password,
 } );
 
+const checkEmailAvailability = async ( emailAddress ) => {
+	try {
+		await wp.undocumented().getTitanEmailAddressAvailability( emailAddress );
+		return true;
+	} catch ( e ) {
+		return e.statusCode !== 409;
+	}
+};
+
+const decorateMailboxWithExistenceError = ( mailbox ) => {
+	mailbox.mailbox.error = translate( '{{strong}}%(email)s{{/strong}} already exists.', {
```

Comment:
I think this would be clearer if we simply state the address is already in use (which is different than it already existing). I've also added a comment to the back end API review to make it easier to identify the existing mailbox/forward/group type.
```suggestion
	mailbox.mailbox.error = translate( '{{strong}}%(email)s{{/strong}} is already in use.', {
```
I also think we should add a translator comment to indicate that `email` is a partial email address.